contracts/pcv/angle/AngleUniswapPCVDeposit.sol

```diff
@@ -0,0 +1,212 @@
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function stake(uint256 amount) external;
+
+    function withdraw(uint256 amount) external;
+
+    function getReward() external;
+}
+
+/// @title implementation for Angle PCV Deposit
+/// @author Angle Core Team and Fei Protocol
+contract AngleUniswapPCVDeposit is UniswapPCVDeposit {
+    using Decimal for Decimal.D256;
+
+    /// @notice the Angle StableMaster contract
+    IStableMaster public immutable stableMaster;
+
+    /// @notice the Angle PoolManager contract
+    IPoolManager public poolManager;
+
+    /// @notice the Angle StakingRewards contract
+    IStakingRewards public stakingRewards;
+
+    /// @notice Uniswap PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _pair Uniswap Pair to deposit to
+    /// @param _router Uniswap Router
+    /// @param _oracle oracle for reference
+    /// @param _backupOracle the backup oracle to reference
+    /// @param _maxBasisPointsFromPegLP the max basis points of slippage from peg allowed on LP deposit
+    constructor(
+        address _core,
+        address _pair,
+        address _router,
+        address _oracle,
+        address _backupOracle,
+        uint256 _maxBasisPointsFromPegLP,
+        IStableMaster _stableMaster,
+        IPoolManager _poolManager,
+        IStakingRewards _stakingRewards
+    ) UniswapPCVDeposit(_core, _pair, _router, _oracle, _backupOracle, _maxBasisPointsFromPegLP) {
+        stableMaster = _stableMaster;
+        poolManager = _poolManager;
+        stakingRewards = _stakingRewards;
+        require(_poolManager.token() == address(fei()), "AngleUniswapPCVDeposit: invalid poolManager");
+        require(_stableMaster.agToken() == token, "AngleUniswapPCVDeposit: invalid stableMaster");
+        require(_stakingRewards.stakingToken() == _pair, "AngleUniswapPCVDeposit: invalid stakingRewards");
+
+        // Approve FEI on StableMaster to be able to mint agTokens
+        SafeERC20.safeApprove(IERC20(address(fei())), address(_stableMaster), type(uint256).max);
+        // Approve LP tokens on StakingRewards to earn ANGLE rewards
+        SafeERC20.safeApprove(IERC20(_pair), address(_stakingRewards), type(uint256).max);
+    }
+
+    /// @notice claim staking rewards
+    function claimRewards() external {
+        stakingRewards.getReward();
+    }
+
+    /// @notice mint agToken from FEI held on this contract
+    /// @dev the call will revert if slippage is too high compared to oracle.
+    function mintAgToken(uint256 amountFei)
+        public
+        onlyPCVController
+    {
+        uint256 minAgTokenOut = Decimal.from(amountFei)
+          .div(readOracle())
+          .mul(Constants.BASIS_POINTS_GRANULARITY - maxBasisPointsFromPegLP)
+          .div(Constants.BASIS_POINTS_GRANULARITY)
+          .asUint256();
+
+        uint256 agTokenBalanceBefore = IERC20(token).balanceOf(address(this));
+        stableMaster.mint(
```

Comment:
should this contract be able to mint FEI?

Reply:
it reduces the attack surface on the protocol if we just pre-mint FEI in the proposal & use it here, but we could mint in this function & grant minter in the proposal. Since it's a one-off, I don't think it's a good idea, though.

Reply:
Does it not already have minter?

Reply:
ah yes... stupid me

Reply:
I'll update to mint the FEI in this function